flowmachine/flowmachine/core/server/query_schemas/base_exposed_query.py

```diff
@@ -0,0 +1,122 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from abc import ABCMeta, abstractmethod
+from copy import deepcopy
+
+from flowmachine.core import Query
+from flowmachine.core.query_info_lookup import QueryInfoLookup
+
+__all__ = ["BaseExposedQuery"]
+
+
+class BaseExposedQuery(metaclass=ABCMeta):
```

Comment:
I need to revisit why there's this distinction, I did understand it but the reason is currently lost to me.